### PR TITLE
API consistency: switch widget

### PR DESCRIPTION
**Pull request by bruno-rino**
The Switch widget API becomes more consistent with other widgets by renaming some properties:
- label -> text
- is_on -> value
- on_toggle -> on_change

The legacy API still works, with deprecation warnings. Exceptions are raised if both the new and old init parameters are used. The label text parameter, now called `text`, was temporarily changed to keyword argument. It should be changed back to positional argument when the compatibility code is removed.

## PR Checklist:
- [x] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

File: src/core/toga/widgets/switch.py
Comment (freakboy3742) [inline src/core/toga/widgets/switch.py:43]: This approach is good; but we should also include:
1. A code comment describing that this is a backward compatibility shim
2. Error handling for the case where someone sets *both* `on_toggle` and `on_change`. 

Even better would be to make the "backwards compatibility" portion of the code as simple as possible, leaning into the "future state" - that is, make the backwards compatibility code something that can be cleanly deleted, rather than something where the logic will need to be re-written. 
```
# 2022-07: Backwards compatibility: on_toggle replace with on_change
if on_toggle:
    if on_change:
        raise error
    else:
        on_change = on_toggle
# end backwards compatibility.

self.on_change = on_change
```
That way, the future code change is a clean deletion of 7 lines of code, rather than a complex revision.

File: src/core/toga/widgets/switch.py
Comment (freakboy3742) [inline src/core/toga/widgets/switch.py:53]: This needs a backwards compatibility path, too.

File: src/core/toga/widgets/switch.py
Comment (freakboy3742) [inline src/core/toga/widgets/switch.py:49]: Same approach is needed here as with on_toggle.

File: src/core/toga/widgets/switch.py
Comment (freakboy3742) [reply src/core/toga/widgets/switch.py:53]: I see this is further down. It might be a good idea to capture all the backwards compatibility code in a single block - again, so the future edit is "delete a single block of N lines", rather than a complex audit.

File: src/core/toga/widgets/switch.py
Comment (freakboy3742) [inline src/core/toga/widgets/switch.py:41]: It would be fairly straightforward to add support for a named `label=` argument in the constructor,  so I'd argue we should add this as well. Yes, it means introducing a "temporary" keyword argument, but it's also an easy thing to delete in future. We can also re-group the arguments so there's a single block of "deprecated" arguments at the end.

**Review by freakboy3742 — CHANGES_REQUESTED** [review]
For the most part, this looks great - it's all fairly mechanical renames in the demos and the backends. The only real comments I have are in the core interface - and those pretty much all relate to making it easy for the future PR that removes the backwards compatibility shim.

**Comment by bruno-rino** [comment]
It's my first time updating a PR. Let me know if I did something wrong.

**Comment by bruno-rino** [comment]
I'm not clear on how to move forward. Should I "Resolve Conversation", to signal that I addressed the changes requested? When these changes are OK, I'll carry on with the other changes laid out in https://github.com/beeware/toga/discussions/1521. Should I open new PR for those, or append to this PR?

**Comment by freakboy3742** [comment]
@bruno-rino Apologies for not responding earlier - I missed the update when you pushed the extra code. As an aside - if you've got a PR where you've made an update and you've been waiting for a review and haven't heard anything for a couple of days, don't be afraid to give a gentle nudge. Sometimes there will be a reason that we haven't responded, but more often than not, it's a simple case of missing a notification that a PR update is ready for review.

Generally speaking, we prefer if the person who opens the conversation resolves it (on the basis that the person who raises a point is the person who can judge if the point has been addressed).

In terms of PR process - the general guidance is "one idea per PR", and try to keep that idea as small as possible. Keeping the PRs small makes them easier to review, which means the review/merge cycle is faster, and there's less chance of a PR turning into a multi-month epic with weekly merge collisions to resolve. In this case "clean up the Switch widget" is a good target"; Something like "use `.value` consistently" would probably be on the larger side, but probably just under the "one idea" limit. A single "make the Toga API consistent" PR would be way too large to be practical.

File: src/core/toga/widgets/switch.py
Comment (freakboy3742) [inline src/core/toga/widgets/switch.py]: We can add that defence though; you use a placeholder other than None, and check for that.

**Review by freakboy3742 — APPROVED** [review]
I've made a couple of minor cosmetic tweaks, including addressing the problem around detecting is_on and value both being specified, and a couple of old API uses in the demos - but otherwise, this looks great! Thanks for the contribution!

**Comment by freakboy3742** [comment]
The test failure is caused by https://github.com/sphinx-doc/sphinx/issues/10710 - a fix for that should land very soon. I'll force a re-run of the test suite once that lands.

**Pull request merged**
